Toggl.Foundation.Tests/MvvmCross/ViewModels/SelectProjectViewModelTests.cs

```diff
@@ -606,79 +565,103 @@ public async Task SortsProjectsByName()
+                var observer = TestScheduler.CreateObserver<IEnumerable<CollectionSection<string, AutocompleteSuggestion>>>();
+                ViewModel.Suggestions.Subscribe(observer);
 
+                await ViewModel.Initialize();
                 TestScheduler.Start();
 
-                ViewModel.ToggleTaskSuggestionsCommand.Execute((ProjectSuggestion)ViewModel.Suggestions[0][1]);
+                ViewModel.ToggleTaskSuggestions.Execute(suggestions[0]);
+                TestScheduler.Start();
 
-                ViewModel.Suggestions.Should().HaveCount(1);
-                ViewModel.Suggestions[0].Should().HaveCount(5);
-                tasksNames(ViewModel.Suggestions[0].Skip(2)).Should().BeInAscendingOrder();
+                var latestSuggestions = observer.LastEmittedValue();
+                latestSuggestions.Should().HaveCount(1);
+                latestSuggestions.First().Items.Should().HaveCount(5);
+                latestSuggestions.First().Items
+                    .Where(suggestion => suggestion is TaskSuggestion)
+                    .Cast<TaskSuggestion>()
+                    .Should()
+                    .BeInAscendingOrder(taskSuggestion => taskSuggestion.Name);
             }
 
             [Fact, LogIfTooSlow]
-            public async Task DoesNotContainSelectedProjectIfProjectIdIsNull()
+            public async Task DoesNotContainASelectedProjectIfProjectIdIsNull()
             {
-                prepareProjects();
+                var suggestions = getProjectSuggestions(20, 10);
+                var suggestionsObservable = Observable.Return(suggestions);
+                InteractorFactory
+                    .GetProjectsAutocompleteSuggestions(Arg.Any<IList<string>>())
+                    .Execute()
+                    .Returns(suggestionsObservable);
                 var parameter = SelectProjectParameter.WithIds(null, null, 0);
+                var observer = TestScheduler.CreateObserver<IEnumerable<CollectionSection<string, AutocompleteSuggestion>>>();
+                ViewModel.Suggestions.Subscribe(observer);
+
                 ViewModel.Prepare(parameter);
                 await ViewModel.Initialize();
                 TestScheduler.Start();
 
-                ViewModel.Suggestions.Should().HaveCount(1);
-                ViewModel.Suggestions.First().Should().OnlyContain(
-                    suggestion => ((ProjectSuggestion)suggestion).Selected == false);
+                var latestSugestions = observer.Messages.Last().Value.Value;
+                latestSugestions.Should().HaveCount(1);
+                latestSugestions.First().Items
+                    .Cast<ProjectSuggestion>()
+                    .Should().OnlyContain(suggestion => !suggestion.Selected);
             }
 
             [Fact, LogIfTooSlow]
             public async Task ContainsOnlyOneSelectedProjectIfProjectIdIsSet()
             {
-                prepareProjects();
+                var suggestions = getProjectSuggestions(20, 10);
```

Comment:
```suggestion
                var suggestions = getProjectSuggestions(20, workspaceId: 10);
```
This would make it easier to understand the tests as well. It's not obvious when the second argument is not a named variable.

Also `getProjectSuggestions` would've been better named: `generateProjectSuggestions` or `createProjectSuggestions` or similar